include/wdl.h

```diff
@@ -265,6 +265,8 @@ WD_HIMAGE wdLoadImageFromFile(const WCHAR* pszPath);
 WD_HIMAGE wdLoadImageFromIStream(IStream* pStream);
 WD_HIMAGE wdLoadImageFromResource(HINSTANCE hInstance,
                 const WCHAR* pszResType, const WCHAR* pszResName);
+WD_HIMAGE wdCreateImageFromBuffer(UINT uWidth, UINT uHeight, const BYTE* pBuffer, 
+                BOOL bHasAlpha, const COLORREF* cPalette);
```

Comment:
Would you mind to change the interface to something more extensible in the future?
I think about something like this:

``` C
#define WD_PIXELFORMAT_PALETTE     1  // cPaletter is used
#define WD_PIXELFORMAT_R8G8B8      2  // RGB without alpha
#define WD_PIXELFORMAT_R8G8B8A8    3  // RGB with alpha
....  // or enum or something like that what describes what kind of data the pBuffer contains.

WD_HIMAGE wdCreateImageFromBuffer(UINT uWidth, UINT uHeight, UINT uStride, const BYTE* pBuffer, 
                int pixelFormat, const COLORREF* cPalette, UINT uPaletteSize);
```

Note I also added argument `uPaletteSize`. It would allow nice optimization in `wdBufferMap2Bitmap()` by re-computing the palette into target format and then copy DWORDs, not BYTEs. And it is imho also better for debugging when anything goes wrong.

Reply:
Just to prevent misunderstanding, I do not ask for support for more pixel formats then you do. But I want to keep the door opened in such need arises in the future.

Reply:
Internally I don't use uPaletteSize for nothing. I don't think it is worth it to optimize that DWORD copy, and it is only possible when destination is 32 bpp, we will still need a byte by byte copy code for 24 bpp used in GD+.